include/NeoFOAM/fields/field.hpp

```diff
@@ -66,6 +66,24 @@ class Field
         data_ = static_cast<ValueType*>(ptr);
     }
 
+    /**
+     * @brief Create a Field with a given size on an executor and uniform value
+     * @param exec  Executor associated to the matrix
+     * @param size  size of the matrix
+     */
+    Field(const Executor& exec, size_t size, ValueType value)
+        : size_(size), data_(nullptr), exec_(exec)
+    {
+        void* ptr = nullptr;
+        std::visit(
+            [this, &ptr, size](const auto& concreteExec)
+            { ptr = concreteExec.alloc(size * sizeof(ValueType)); },
```

Comment:
can this be renamed?
```suggestion
            [this, &ptr, size](const auto& exec)
            { ptr = exec.alloc(size * sizeof(ValueType)); },
```